app/src/main/java/de/westnordost/streetcomplete/MainActivity.kt

```diff
@@ -23,7 +23,6 @@ import androidx.localbroadcastmanager.content.LocalBroadcastManager
 import androidx.preference.PreferenceManager
 import de.westnordost.osmapi.common.errors.OsmApiException
 import de.westnordost.osmapi.common.errors.OsmApiReadResponseException
-import de.westnordost.osmapi.common.errors.OsmAuthorizationException
 import de.westnordost.osmapi.common.errors.OsmConnectionException
```

Comment:
Here

Reply:
These exceptions are not documented on the `MapDataApi` and `NotesApi` from the de.westnordost.osmapi package because they are like "unchecked exceptions", see line 256ff

I'd say, wrap em all in "ConnectionException" or so. These kind of errors should just lead to a toast message displayed to the user that upload/download is currently not possible.
Since these errors could be thrown on any API call (f.e. if the OpenStreetMap API is down for maintenance, connection is lost due to bad network etc), I fear these need to be caught on every single API function. Maybe write an inline helper function like `inline fun tryConnect(block: () -> T): T` so to not bloat up the code there.
Maybe it also makes sense to not catch the individual erros but have an `catch(e: Exception)` at the end?